packages/react/src/components/FocusTrapZone/FocusTrapZone.tsx

```diff
@@ -97,284 +93,231 @@ export const FocusTrapZone: React.FunctionComponent<IFocusTrapZoneProps> & {
+   */
+  const focusFTZ = useEventCallback(() => {
+    if (!root.current) {
+      return; // not done mounting
+    }
+
+    const { previouslyFocusedElementInTrapZone } = internalState;
     if (
       focusPreviouslyFocusedInnerElement &&
-      internalState.previouslyFocusedElementInTrapZone &&
-      elementContains(root.current, internalState.previouslyFocusedElementInTrapZone)
+      previouslyFocusedElementInTrapZone &&
+      elementContains(root.current, previouslyFocusedElementInTrapZone)
     ) {
       // focus on the last item that had focus in the zone before we left the zone
-      focusAsync(internalState.previouslyFocusedElementInTrapZone);
+      focusElementAsync(previouslyFocusedElementInTrapZone);
       return;
     }
 
-    const focusSelector =
-      typeof firstFocusableSelector === 'string'
-        ? firstFocusableSelector
-        : firstFocusableSelector && firstFocusableSelector();
-
     let firstFocusableChild: HTMLElement | null = null;
 
-    if (root.current) {
-      if (typeof firstFocusableTarget === 'string') {
-        firstFocusableChild = root.current.querySelector(firstFocusableTarget);
-      } else if (firstFocusableTarget) {
-        firstFocusableChild = firstFocusableTarget(root.current);
-      } else if (focusSelector) {
-        firstFocusableChild = root.current.querySelector('.' + focusSelector);
-      }
+    if (typeof firstFocusableTarget === 'string') {
+      firstFocusableChild = root.current.querySelector(firstFocusableTarget);
+    } else if (firstFocusableTarget) {
+      firstFocusableChild = firstFocusableTarget(root.current);
+    } else if (firstFocusableSelector) {
+      const focusSelector =
+        typeof firstFocusableSelector === 'string' ? firstFocusableSelector : firstFocusableSelector();
+      firstFocusableChild = root.current.querySelector('.' + focusSelector);
+    }
 
-      // Fall back to first element if query selector did not match any elements.
-      if (!firstFocusableChild) {
-        firstFocusableChild = getNextElement(
-          root.current,
-          root.current.firstChild as HTMLElement,
-          false,
-          false,
-          false,
-          true,
-        );
-      }
+    // Fall back to first element if query selector did not match any elements.
+    if (!firstFocusableChild) {
+      firstFocusableChild = getNextElement(
+        root.current,
+        root.current.firstChild as HTMLElement,
+        false,
+        false,
+        false,
+        true,
+      );
     }
+
     if (firstFocusableChild) {
-      focusAsync(firstFocusableChild);
+      focusElementAsync(firstFocusableChild);
     }
-  }, [firstFocusableSelector, firstFocusableTarget, focusPreviouslyFocusedInnerElement, internalState]);
+  });
 
-  const onBumperFocus = React.useCallback(
-    (isFirstBumper: boolean) => {
-      if (disabled) {
-        return;
-      }
+  /** Used in root div focus/blur handlers */
+  const focusBumper = (isFirstBumper: boolean) => {
```

Comment:
This no longer uses useCallback because that's not necessary for event handlers passed to native elements